runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -240,6 +264,23 @@ else if (state == GameState.LOGIN_SCREEN)
 		}
 	}
 
+	private void updateAttackStyle(int attackStyleIndex, int castingMode)
```

Comment:
If anything, I would expect most of this implementation to be a static method of `AttackStyle` to get the attack style for some given attack style and casting mode vars.

Reply:
Hmm, I'm not sure if I'm following. Something like this?

```java
// XpTrackerPlugin.java
private void updateAttackStyle(int attackStyleIndex, int castingMode)
{
    attackStyle = AttackStyle.getByWeaponType(
        attackStyleIndex,
        equippedWeaponType,
        castingMode);
}

// AttackStyle.java
public static AttackStyle getByWeaponType(int attackStyleIndex, WeaponType equippedWeaponType, int castingMode) {
    AttackStyle attackStyle;
    AttackStyle[] attackStyles = equippedWeaponType.getAttackStyles();

    if (attackStyleIndex < attackStyles.length)
    {
        attackStyle = attackStyles[attackStyleIndex];
        if (attackStyle == null)
        {
            attackStyle = OTHER;
        }
        else if ((attackStyle == CASTING) && (castingMode == 1))
        {
            attackStyle = DEFENSIVE_CASTING;
        }
    }

    return attackStyle;
}
```

The tradeoff would be this class wouldn't have to import `CASTING`, `DEFENSIVE_CASTING`, and `OTHER` but `AttackStyle` would have to handle values without much context. It's pretty [bare bones](https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStyle.java?w=1#L30) right now and isn't associated with weapons.

Reply:
After thinking about this through, it might make more sense to have this logic live in `WeaponType`. It's already importing `CASTING`, `DEFENSIVE_CASTING`, and `OTHER`